### ISSUE TITLE
Oddly Visible Fleet

### ISSUE THREAD

**Issue by geoffthemedio** [freeorion/freeorion#1478]
In the attached zipped save, there is a fleet that appears in otherwise empty space. By cheating with detection range and loading an autosave, it seems that it is on a starlane that I haven't yet directly seen either end off. Due to not knowing the lane geometry, the ship is rendered a bit oddly, with the icon oriented vertically. It might be better to render the fleet somewhat differently so that it's clearer that it's moving, though I'm not sure if or how this should be done.

Not sure if this is a bug, or is fine as is... but thought I should mention it.

![odd_fleet](https://cloud.githubusercontent.com/assets/11386908/24831514/1fb183f6-1c9b-11e7-9e3a-d2a70cd9989a.png)

![cheaty](https://cloud.githubusercontent.com/assets/11386908/24831569/85255acc-1c9c-11e7-8513-04dd7d4e9a6a.png)

[oddly_visible_troop_ships.zip](https://github.com/freeorion/freeorion/files/907731/oddly_visible_troop_ships.zip)

**Comment by Dilvish-fo** [comment]
I've noticed that sort of thing before.  Although it looks odd, I don't think it's a bug.  I don't really see anything better to do about it, other than perhaps add a mention of the possibility in some part of the Pedia, like in a discussion of Starlanes or Visibility.

**Comment by l29ah** [comment]
By the way, a fleet on a vanished starlane is directed properly.

**Comment by Dilvish-fo** [comment]
>  It might be better to render the fleet somewhat differently so that it's clearer that it's moving, 

It does seem likely to me that rotating the fleet icon 90 degrees, so that it is pointed sideways, might give more of the impression of movement.

**Comment by dbenage-cx** [comment]
Assuming the empire may not know if the ship is moving, or which specific direction it is heading.
Rotating the icon may still give the false impression of a known direction, similar to the current situation.
Maybe have some directional indicator within the selection circle or an alternate stationary fleet icon (which could be used when unknown as well).

**Comment by LGM-Doyle** [comment]
Another solution is to not render ships on starlanes that have not been detected yet.  If an empire hasn't seen the starlane, then it can't see the ship.

**Comment by MatGB** [comment]
I'd rather not go that way, if something is in range yet should see it, better to reveal the lane than not reveal the ship, especially with starlane removal effects being more common.

**Comment by LGM-Doyle** [comment]
I wrote "have not been detected yet".  I did not mean for ships to vanish if the starlane was removed, just that you would you would not "discover" the starlane from the laneless ship.  The other way works as well.

I think the problem is that you should see either both the ship and the starlane, or neither the ship nor the starlane, then there is no ambiguity/weirdness with respect to the ship facing.

**Comment by Dilvish-fo** [comment]
> I think the problem is that you should see either both the ship and the starlane, or neither the ship nor the starlane,

I do not share that sentiment, and would object to the latter prong of hiding ships just because their starlane was unknown.  

I am not sure I object  to the first prong.  Visibility of a moving ship could conceivably grant Basic Visibility of the previous and next systems and the associated starlane, but that would be giving more info than we do currently and I just don't think it's needed. 

I don't think it's necessary to do anything new to the ship icon either, because then you'd still have to educate people about the meaning of whatever it is you settle on doing to the icon, and you may as well just educate them that ship icons floating in the middle of nowhere are on a not-yet-known starlane and are always represented with a vertical alignment regardless of their actual direction of motion.

**Comment by LGM-Doyle** [comment]
I agree about the education, which is why I lean towards show neither.

In the case of a not-yet-known starlane, the ship's exit from the starlane will also not be observed.  So the unoriented ship icon will go from visible in the middle of nowhere to ghosted in the middle of nowhere.  This doesn't help and looks buggy.

The fluff explanation for why you see neither the ship nor the starlane, if you have never observed either endpoint could be that with faster than light (FTL) technology you can't observe ship in a starlane unless you have observed the spacetime anomalies at either of the endpoints.

A method to education might be to change the title text in the `FleetWnd` from "Fleets near X" to Fleets on unknown starlane" and on the individual fleets from "Moving to X" to "Moving to unknown system".

**Comment by Dilvish-fo** [comment]
> Visibility of a moving ship could conceivably grant Basic Visibility of the previous and next systems and the associated starlane, but that would be giving more info than we do currently and I just don't think it's needed.

This issue has lingered around quite a while, I think we ought to be able to get it resolved, either (a) simply accepting the current handling, or else (b) having visibility of a fleet grant knowledge of the starlane it is on as well as basic visibility of the systems at either end.  I had initially been somewhat opposed to the latter, but over time I find myself slightly favoring the granting of starlane knowledge.

@geoffthemedio @Vezzra @MatGB could you guys please vote a or b?

**Comment by MatGB** [comment]
B for me out of preference but I'm not that bothered

**Comment by geoffthemedio** [comment]
No strong opinion...

**Comment by LGM-Doyle** [comment]
I still prefer (c) without visibility of the starlane the ship is not visible.

**Comment by Vezzra** [comment]
I don't really like (b), so my preference would be (a), although I can see the issue with that. The best solution IMO would be to render the icon with correct orientation, so it looks less buggy/weird, but apparently that would require some non-trivial code changes (basically the orientation would have to become a property of the fleet, so that info can be passed to the client).

However, I guess I can live with (b), so if you guys prefer that, go ahead.
